File: languages/go/entity/identity/did.go
package identity

// Did represents a DID.
type Did struct {
	Did       string
	DidMethod DidMethod
}

// NewDid returns a new instance of Did for the given parameters.
func NewDid(did string, didMethod DidMethod) Did {
	return Did{
		Did:       did,
		DidMethod: didMethod,
	}
}

File: languages/js/src/entity/identity/did.ts
import { DidMethod } from "./did_method";

/**
 * Represents parameters used for generating DIDs.
 */
export class Did {
    did: string;
    didMethod: DidMethod;

    /**
     * Constructs a Did object with the specified parameters.
     * @param did 
     * @param didType 
     */
    constructor(did: string, didMethod: DidMethod) {
        this.did = did;
        this.didMethod = didMethod;
    }
}